barretenberg/cpp/src/barretenberg/dsl/acir_format/ec_operations.cpp

```diff
@@ -35,6 +65,8 @@ void create_ec_add_constraint(Builder& builder, const EcAdd& input, bool has_val
     } else {
         builder.assert_equal(y_normalized.witness_index, input.result_y);
     }
+    // TODO: remove the infinite result, because the function should always return a non-zero point.
+    // But this requires an ACIR serialisation change and it will be done in a subsequent PR.
```

Comment:
From this comment it looks like the ACIR representation of an ecc point doesn't allow for a point at infinity?

What happens if I have two constant points, `A` and `B` where `A = -B`, and I compute `A+B` in noir? Is the compiler clever enough to return a point at infinity, or will the compiler cause the current code path (i.e. `acir_format::create_ec_add_constraint`) to be executed?

If it's the latter then I think it's better if we throw an error for the case where `infinite.is_constant()` is triggered, because the alternative is weird undefined behavior. If we're planning to fix the ACIR serialization in the near term we can skip, but if we forget about this issue it will cause us problems in the future because, if I understand correctly, the current implementation produces either undefined behavior or underconstrained code (i.e. the case where we add a constant point with its inverse)